src/local_scheduler/local_scheduler.cc

```diff
@@ -728,9 +831,15 @@ void process_message(event_loop *loop,
   case MessageType_GetTask: {
     /* If this worker reports a completed task: account for resources. */
     if (worker->task_in_progress != NULL) {
-      TaskSpec *spec = Task_task_spec(worker->task_in_progress);
-      /* Return dynamic resources back for the task in progress. */
-      update_dynamic_resources(state, spec, true);
+      /* Return dynamic resources back for the task in progress. If the worker
+       * is an actor, then we do not return GPU resources. */
+      if (ActorID_equal(worker->actor_id, NIL_ACTOR_ID)) {
+        release_resources(state, worker, (double) worker->cpus_in_use,
+                          (double) worker->gpus_in_use.size());
+      } else {
+        release_resources(state, worker, (double) worker->cpus_in_use, 0);
+      }
```

Comment:
We could simplify this even further by calling `release_resources` on the spec for the `task_in_progress`.